sdk/python/foundation-models/system/content-safety/score/score.py

```diff
@@ -0,0 +1,323 @@
+
+
+def init():
+    # check if aacs access key is provided
+    get_aacs_access_key()
+    _logger.info("init")
+
+
+def analyze_text(text):
+    endpoint = os.environ.get("CONTENT_SAFETY_ENDPOINT")
+    key = get_aacs_access_key()
+
+    # Create an Content Safety client
+    client = ContentSafetyClient(endpoint, AzureKeyCredential(key))
+
+    # Build request
+    request = AnalyzeTextOptions(text=text)
+
+    # Analyze text
+    try:
+        response = client.analyze_text(request)
+    except Exception as e:
+        raise e
+
+    severity = 0
+
+    if response.hate_result is not None:
+        _logger.info("Hate severity: {}".format(response.hate_result.severity))
+        severity = max(severity, response.hate_result.severity)
+    if response.self_harm_result is not None:
+        _logger.info("SelfHarm severity: {}".format(response.self_harm_result.severity))
+        severity = max(severity, response.self_harm_result.severity)
+    if response.sexual_result is not None:
+        _logger.info("Sexual severity: {}".format(response.sexual_result.severity))
+        severity = max(severity, response.sexual_result.severity)
+    if response.violence_result is not None:
+        _logger.info("Violence severity: {}".format(response.violence_result.severity))
+        severity = max(severity, response.violence_result.severity)
+
+    return severity
+
+
+@input_schema("input_data", input_param)
+@output_schema(output_param)
+def run(input_data):
+    if (
+        isinstance(input_data, np.ndarray)
+        or (
+            isinstance(input_data, dict)
+            and input_data
+            and isinstance(list(input_data.values())[0], np.ndarray)
+        )
+        or (pandas_installed and isinstance(input_data, pd.DataFrame))
+    ):
+        result = model.predict(input_data)
+
+        return _get_jsonable_obj(result, pandas_orient="records")
+
+    # Format input
+    if isinstance(input_data, str):
+        input_data = json.loads(input_data)
+    if "input_data" in input_data:
+        input_data = input_data["input_data"]
+    if is_hfv2:
+        input = input_data
+    elif isinstance(input_data, list):
+        # if a list, assume the input is a numpy array
+        input = np.asarray(input_data)
+    elif (
+        isinstance(input_data, dict)
+        and "columns" in input_data
+        and "index" in input_data
+        and "data" in input_data
+    ):
+        # if the dictionary follows pandas split column format, deserialize into a pandas Dataframe
+        input = pd.read_json(json.dumps(input_data), orient="split", dtype=False)
+    else:
+        # otherwise, assume input is a named tensor, and deserialize into a dict[str, numpy.ndarray]
+        input = {
+            input_name: np.asarray(input_value)
+            for input_name, input_value in input_data.items()
+        }
+
+    result = model.predict(input)
+    severity = max([analyze_text(row[0]) for _, row in result.iterrows()])
```

Comment:
Let's check if we need to add this for "input" also as mentioned in the group chat